internal/pkg/cli/interfaces.go

```diff
@@ -166,15 +165,19 @@ type secretDeleter interface {
 }
 
 type imageBuilderPusher interface {
-	BuildAndPush(docker repository.ContainerLoginBuildPusher, args *dockerengine.BuildArguments) (string, error)
+	BuildAndPush(args *dockerengine.BuildArguments) (string, error)
 }
-
 type repositoryURIGetter interface {
 	URI() (string, error)
 }
 
+type dockerLogin interface {
```

Comment:
nit: the interface function seems docker-agnostic!

```suggestion
type repositoryLogin interface {
```